src/components/MultiHopTrade/components/TradeInput/hooks/useSharedHieght.tsx

```diff
@@ -24,5 +24,5 @@ export const useSharedHeight = () => {
     }
   }, [])
 
-  return { observedRef, height }
+  return useMemo(() => ({ observedRef, height }), [height])
```

Comment:
Ditto

Reply:
done ser
5e35960fcdd69296fff2972739c36bdf9af68c4c

Reply:
as discussed, memoization is required as the object containing those values is referentially changed